### ISSUE TITLE
Can't use hash character (#) in description field in "add activity" window

### ISSUE THREAD

**Issue by rhertzog** [projecthamster/hamster#757]
When you insert any text containing a hash (`#`) character in the "description" field of the "Add an activity" window, then the "Save" button is greyed out and you can't add the activity.

Note that the hash character doesn't have to be at the start of the word, so it's not only tag-words that are problematic, it's also any URL with an anchor (e.g. `https://example.org/#hello`).

However if you input the description with the hash character within the "cmdline" field then it is accepted... for example cmdline = `09:06 Activity@Category, https://example.org/#hello`.

/cc @Flupp @GeraldJansen

**Comment by rhertzog** [comment]
Note that https://github.com/projecthamster/hamster/pull/755 would change the behaviour here. With this patch applied, the description can contain `https://example.org/#hello` (it is accepted and the save button is not greyed out) but a tag word is still not allowed in the description while it is a syntax that we decided to support.

My first implementation stripped the hash from tag words from the description while adding them into proper tags. But the general feedback was that we should not strip the hash character at all, thus we should allow tag words inside the description.

We should thus fix the behavior of this edit window to actually assign tags when we edit the description field with a tag word inside it.

**Comment by GeraldJansen** [comment]
Harvesting tags from the description field as one types is complicated by the need to keep three fields in sync (cmdline, description and tags). I played with this a bit, including an effort to just sync the fields on focus_out from the description field, but I didn't manage to find a satisfactory solution.

**Comment by rhertzog** [comment]
Yeah, it seems unreasonable to do it while typing because you would add multiple partial tags while your inputting the tag (i.e. typing `#tag` would add `t`, `ta` and `tag`). But doing it on focus_out (+ maybe once before save in case focus_out is not executed when you validate the save button with a keyboard shortcut) should be doable no?

What difficulties did you hit?

**Comment by vossenjp** [comment]
This just bit me.

My use-case is `https://example.com/redmine/issues/1234#note-5` which I do a lot, and I just upgrade to Mint-22.3 and ran into this `hamster-time-tracker-3.0.3-1ubuntu1` issue.

The #755 PR fixes it.  Spoiler:
```
$ diff -u /usr/lib/python3/dist-packages/hamster/lib/parsing.py.ORIG /usr/lib/python3/dist-packages/hamster/lib/parsing.py 
--- /usr/lib/python3/dist-packages/hamster/lib/parsing.py.ORIG	2026-03-03 17:27:01.863850464 -0500
+++ /usr/lib/python3/dist-packages/hamster/lib/parsing.py	2026-03-03 17:40:13.937354696 -0500
@@ -18,7 +18,10 @@
     )
 """, flags=re.VERBOSE)
 
+# Bug: https://github.com/projecthamster/hamster/issues/757
+# PR: https://github.com/projecthamster/hamster/pull/755
 tags_in_description = re.compile(r"""
+    \s
     \#
     (?P<tag>
         [a-zA-Z] # Starts with an alphabetic character (digits excluded)
```
